providers/implementations/keymgmt/slh_dsa_kmgmt.c.in

```diff
@@ -125,7 +131,23 @@ static int slh_dsa_import(void *keydata, int selection, const OSSL_PARAM params[
         return 0;
 
     include_priv = ((selection & OSSL_KEYMGMT_SELECT_PRIVATE_KEY) != 0);
-    return ossl_slh_dsa_key_fromdata(key, p.pub, p.priv, include_priv);
+    res = ossl_slh_dsa_key_fromdata(key, p.pub, p.priv, include_priv);
+#ifdef FIPS_MODULE
+    /*
+     * FIPS 140-3 IG 10.3.A additional comment 1 mandates that a pairwise
+     * consistency check be undertaken on key import.  The required test
+     * is described in SP 800-56Ar3 5.6.2.1.4.
+     */
+    if (res > 0 && ossl_slh_dsa_key_has(key, OSSL_KEYMGMT_SELECT_KEYPAIR) > 0)
+        if (!slh_dsa_fips140_pairwise_test(key, NULL)) {
+            ERR_raise_data(ERR_LIB_PROV, PROV_R_INVALID_KEY,
```

Comment:
Shouldn't be used `ossl_set_error_state(OSSL_SELF_TEST_TYPE_PCT)` instead of raising error?

Reply:
I don't believe so.  Failing a PCT isn't cause to put the module into an error state unless the rules have changed.

Reply:
I've asked the lab for clarification.